apiserver/application/application.go

```diff
@@ -945,6 +945,44 @@ func (api *API) saveRemoteApplication(
 			Scope:     ep.Scope,
 		}
 	}
+
+	// If the a remote application with the same name and endpoints from the same
```

Comment:
can you please make the API call and bits below a separate method, so this just becomes:

```go
remoteApp, err := api.existingRemoteApplication(applicationName)
if err == nil {
    return remoteApp, nil
} else if !errors.IsNotFound(err) {
    return nil, errors.Trace(err)
}
```